src/org/opensolaris/opengrok/util/Executor.java

```diff
@@ -148,7 +147,12 @@ public int exec(final boolean reportExceptions, StreamHandler handler) {
         ProcessBuilder processBuilder = new ProcessBuilder(cmdList);
         final String cmd_str = processBuilder.command().toString();
         final String dir_str;
-        Timer t = new Timer();
+        Timer t = null;
+
+        if (this.timeout != 0) {
+            // invoking the constructor starts the background thread
+            t = new Timer();
```

Comment:
could you move it further down - possibly under the `if this.timeout != 0` check ?